### ISSUE TITLE
Option to use VSCode internal language formatter instead of prettier

### ISSUE THREAD

**Issue by SiNONiMiTY** [sveltejs/language-tools#1707]
### Description

Title says, For me, VSCode HTML formatter does a better job formatting the *.svelte files I have rather than Prettier.

A simple option to choose what formatter is enabled by the extension would be very appreciated.

### Proposed solution

Enable an option to use VSCode's internal formatter rather than being dependent on prettier.

**Comment by dummdidumm** [comment]
Is it possible to do this yourself? I beliefe you can choose through the settings.json which formatter runs for what file. If so, I'm not sure we want to add an option for that.

**Comment by SiNONiMiTY** [comment]
Yes I tried that, but I need to associate .svelte to HTML, and I would lose TS and ESLint features.

**Comment by brunnerh** [comment]
Personally, I think the extension should not use Prettier *at all*.

It is fine to use Prettier in your own personal project or in organizations that one is responsible for, but to have such a restrictive and arbitrary formatter in an extension that is used by many vastly different people and companies, each with their own coding styles is just not good.

The only reasonable course of action if there is some divergence is usually to opt out of formatting completely because there just are not enough ways to configure it otherwise.

**Comment by jasonlyu123** [comment]
This is not a plug-in replacement. So even if we want to do it. It won't be a high-priority one. It would be more possible to happen if someone can help us with it.

**Comment by SiNONiMiTY** [comment]
> This is not a plug-in replacement. So even if we want to do it. It won't be a high-priority one. It would be more possible to happen if someone can help us with it.

I am not familiar with the inner workings of VSCode, but is there any VSCode API that the Svelte plugin can call to use the internal formatter instead of prettier?

**Comment by jasonlyu123** [comment]
Sort of, but not what you think it is. We use some low-level APIs that VSCode also uses. We must glue it together like VSCode HTML features do instead of just calling it. 

We also need to add support for svelte tags and mustache. Some of the svelte syntaxes are also not strict in compliance with HTML.
